avm/res/web/static-site/main.bicep

```diff
@@ -257,6 +267,14 @@ resource staticSite_roleAssignments 'Microsoft.Authorization/roleAssignments@202
   }
 ]
 
+module staticSite_privateDnsZone 'br/public:avm/res/network/private-dns-zone:0.7.0' = if (!empty(privateEndpoints) && createPrivateDnsZone == 'Enabled') {
```

Comment:
Considering the amount of properties one is able to configure in the module I wonder if we may need to introduce a parameterObject for them - like done e.g. for the `nicConfigurations` parameter in the VM module. 

Just thinking out loud here. If you cannot configure it to you needs you're essentially required to idempotently redeploy the DNS Zone after the deployment with another deployment. Maybe that's ok and intended - but wanted to raise the thought :)

Reply:
So on this thought, for a VM the NIC is required, I guess here we do follow a smiliar pattern except the PDNS Zone is only needed for private endpoints to work. Typically the DNS Zone should sit with the rest of the DNS Zone and be configured with a vnet link.

In this current implemntation the app will still not be routable as the DNS Zone has not been linked to a Virtual Network, I need to update this so it accepts some properties. My view is this should be the bare minimal requirements to ensure PE connectivity is established and routable but any extensive configuration should be done in a dedicated module for the staticSite zone. il make an update to this today

Reply:
Additional param added for virtualNetworkResourceId

Reply:
I wonder if that's something we should chat about some time - just to ensure we're all on the same page. The next maintainer call would come to mind 💪

Reply:
can we let this go in the meantime?

Reply:
Define "Let go"? Should we not find together the best solution instead of potentially introducing multiple breaking changes? Hence the suggestion with the Maintainer call where these design questions would usually be brought up.

Reply:
#4345 has been open since Jan 29th, here we are talking about a new resource call which is being updated. I would rather us push this change to resolve the current and very limited issue then look at refinements on a wider basis following this. I am unsure what we have to discuss around this, it should be a simply implementation unless the user brings there own DNS Zone?

Reply:
Thanks for the adjustments @ChrisSidebotham, resolving